packages/klighd-core/src/options/actions.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ *
+ * Copyright 2021 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This code is provided under the terms of the Eclipse Public License (EPL).
+ */
+
+import { Action } from "sprotty";
+import {
+    DisplayedActionData,
+    LayoutOptionUIData,
+    LayoutOptionValue,
+    SynthesisOption,
+    ValuedSynthesisOption,
+} from "./option-models";
+
+/** Request message from the server to update the diagram options widget on the client. */
+export class UpdateOptionsAction implements Action {
+    static readonly KIND = "updateOptions";
+    readonly kind = UpdateOptionsAction.KIND;
+
+    constructor(
+        public readonly valuedSynthesisOptions: ValuedSynthesisOption[],
+        public readonly layoutOptions: LayoutOptionUIData[],
+        public readonly actions: DisplayedActionData[],
+        public readonly modelUri: string
+    ) {}
+
+    /** Type predicate to narrow an action to this action. */
+    static isThisAction(action: Action): action is UpdateOptionsAction {
+        return action.kind === UpdateOptionsAction.KIND;
+    }
+}
+
+/**
+ * Triggers a action from the options that should be performed.
+ * Do not confuse this with PerformActionAction!
+ */
+export class PerformOptionsActionAction implements Action {
```

Comment:
They are not. Previously this action did not exists since it was implemented via [direct method calls](https://git.rtsys.informatik.uni-kiel.de/projects/KIELER/repos/keith/browse/keith-diagram-options/src/browser/diagramoptions-view-contribution.ts#131). The UI called the linked code directly. This triggers a LSP notification with a completely different method than `PerformActionAction`, see [here](https://git.rtsys.informatik.uni-kiel.de/projects/KIELER/repos/keith/browse/keith-diagram-options/src/common/index.ts#27).

Now, the UI dispatches above action and is caught by the `OptionsRegistry` which then dispatches the same LSP notification via the `Connection`, implemented [here](https://github.com/kieler/klighd-vscode/blob/d0a0f6c6d6611fb0f94dc1ac11d189338de1555b/packages/klighd-core/src/options/options-registry.ts#L76-L80). So above action is a client only action that enables, along other actions, a flux-like architecture in the UI/State relationship. Since the actions are completely different, I don't think that they should be combined.